rust/cubesql/cubesql/src/compile/engine/udf.rs

```diff
@@ -1198,6 +1198,95 @@ fn last_day_of_month(y: i32, m: u32) -> u32 {
     NaiveDate::from_ymd(y, m + 1, 1).pred().day()
 }
 
+pub fn create_interval_mul_udf() -> ScalarUDF {
+    let fun = make_scalar_function(move |args: &[ArrayRef]| {
+        assert!(args.len() == 2);
+
+        let multiplicands = downcast_primitive_arg!(args[1], "multiplicand", Int64Type);
+
+        match &args[0].data_type() {
+            DataType::Interval(IntervalUnit::YearMonth) => {
+                let intervals = downcast_primitive_arg!(args[0], "interval", IntervalYearMonthType);
+                let result = intervals
+                    .iter()
+                    .zip(multiplicands.iter())
+                    .map(|values| match values {
+                        (Some(interval), Some(multiplicand)) => {
+                            Some(interval * i32::try_from(multiplicand).ok()?)
+                        }
+                        _ => None,
+                    })
+                    .collect::<IntervalYearMonthArray>();
+                Ok(Arc::new(result))
+            }
+            DataType::Interval(IntervalUnit::DayTime) => {
+                let intervals = downcast_primitive_arg!(args[0], "interval", IntervalDayTimeType);
+                let result = intervals
+                    .iter()
+                    .zip(multiplicands.iter())
+                    .map(|values| match values {
+                        (Some(interval), Some(multiplicand)) => {
+                            let interval_value: u64 = interval as u64;
+                            let days: i32 = ((interval_value & 0xFFFFFFFF00000000) >> 32) as i32;
+                            let milliseconds: i32 = (interval_value & 0xFFFFFFFF) as i32;
+                            let multiplicand = i32::try_from(multiplicand).ok()?;
+                            let days_product = days * multiplicand;
+                            let milliseconds_product = milliseconds * multiplicand;
+                            let interval_product = (((days_product as u64) << 32)
+                                | (milliseconds_product as u64))
+                                as i64;
+                            Some(interval_product)
+                        }
+                        _ => None,
+                    })
+                    .collect::<IntervalDayTimeArray>();
+                Ok(Arc::new(result))
+            }
+            _ => Err(DataFusionError::Execution(
+                "unsupported interval type".to_string(),
+            )),
+        }
+    });
+
+    let return_type: ReturnTypeFunction = Arc::new(move |arg_types| {
+        if arg_types.len() != 2 {
```

Comment:
Simple `assert` should be enough. It looks useless because DF will check return_type after validating signature where we define 2 arguments. Any thoughts?

Reply:
Oh, I see. Let's keep it here. It looks like we need to do the same behavior for all return_type functions, but it's a separate task.